edsdme/scripts/personalization.js

```diff
@@ -0,0 +1,81 @@
+import {
+  getCurrentProgramType,
+  getPartnerDataCookieValue,
+  isMember,
+  partnerIsSignedIn,
+  getPartnerDataCookieObject,
+  signedInNonMember,
+  isResseler,
+}
+  from './utils.js';
+
+const PAGE_PERSONALIZATION_PLACEHOLDERS = { firstName: '//*[contains(text(), "$firstName")]' };
+
+const LEVEL_CONDITION = 'partner-level';
+const PERSONALIZATION_MARKER = 'partner-personalization';
+const PROGRAM = getCurrentProgramType();
+const PARTNER_LEVEL = getPartnerDataCookieValue(PROGRAM, 'level');
+const COOKIE_OBJECT = getPartnerDataCookieObject(PROGRAM);
+const PERSONALIZATION_CONDITIONS = {
+  'partner-not-member': signedInNonMember(),
+  'partner-not-signed-in': !partnerIsSignedIn(),
+  'partner-all-levels': isMember(),
+  'partner-reseller': isResseler(PARTNER_LEVEL),
+  'partner-level': (level) => PARTNER_LEVEL === level,
+};
+
+function getNodesByXPath(query, context = document) {
+  const nodes = [];
+  const xpathResult = document.evaluate(query, context);
+  let current = xpathResult?.iterateNext();
+  while (current) {
+    nodes.push(current);
+    current = xpathResult.iterateNext();
+  }
+  return nodes;
+}
+
+function personalizePlaceholders(placeholders, context = document) {
+  Object.entries(placeholders).forEach(([key, value]) => {
+    const placeholderValue = COOKIE_OBJECT[key];
+    getNodesByXPath(value, context).forEach((el) => {
+      if (!placeholderValue) el.remove();
+      el.textContent = el.textContent.replace(`$${key}`, placeholderValue);
+    });
+  });
+}
+
+function removeElement(element, conditions) {
+  if (!element || !conditions?.length) return;
+  const remove = conditions.every((condition) => {
+    const conditionLevel = condition.startsWith(LEVEL_CONDITION) ? condition.split('-').pop() : '';
+    return conditionLevel
+      ? !PERSONALIZATION_CONDITIONS[LEVEL_CONDITION](conditionLevel) : !PERSONALIZATION_CONDITIONS[condition];
+  });
+  if (remove) element.remove();
```

Comment:
does removing rather than hiding have a performance impact?
I guess removing is fine if we get it right the first time :-)
If conditions can change while the user is on the page, it may be better to show/hide

Reply:
I will change the logic so we use css class to hide element instead of removing it